clustering/singleton/extension/src/main/java/org/wildfly/extension/clustering/singleton/SingletonDeploymentResourceDefinition.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2018, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.wildfly.extension.clustering.singleton;
+
+import org.jboss.as.clustering.controller.ChildResourceDefinition;
+import org.jboss.as.clustering.controller.ManagementResourceRegistration;
+import org.jboss.as.clustering.controller.MetricHandler;
+import org.jboss.as.controller.ModelVersion;
+import org.jboss.as.controller.PathElement;
+import org.jboss.as.controller.transform.description.ResourceTransformationDescriptionBuilder;
+import org.jboss.msc.service.ServiceName;
+
+/**
+ * @author Paul Ferraro
+ */
+public class SingletonDeploymentResourceDefinition extends ChildResourceDefinition<ManagementResourceRegistration> {
+
+    static final PathElement WILDCARD_PATH = pathElement(PathElement.WILDCARD_VALUE);
+
+    static PathElement pathElement(ServiceName name) {
+        return pathElement(name.getParent().getSimpleName());
+    }
+
+    static PathElement pathElement(String name) {
+        return PathElement.pathElement("deployment", name);
+    }
+
+    static void buildTransformation(ModelVersion version, ResourceTransformationDescriptionBuilder parent) {
+        if (SingletonModel.VERSION_3_0_0.requiresTransformation(version)) {
+            parent.discardChildResource(WILDCARD_PATH);
```

Comment:
`VERSION_2_0_0` rejects the children, is it safe to discard them here or should they also be rejected?

Reply:
@jamezp Where do you see children being rejected in VERSION_2_0_0?

Reply:
Never mind. Kabir pointed out this is a runtime resource so it's probably okay.